src/lib.rs

```diff
@@ -842,6 +842,30 @@ impl Connection {
         Ok(())
     }
 
+    /// Flush a NATS connection by sending a `PING` protocol and waiting for the responding `PONG`.
+    /// Will fail with `TimedOut` if the server takes longer than this duration to respond.
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> std::io::Result<()> {
+    /// # let nc = nats::connect("demo.nats.io")?;
+    /// nc.flush()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn flush_timeout(&self, duration: Duration) -> io::Result<()> {
```

Comment:
We should have a pain flush as well with default timeout like the go client.

Also, if the client gets disconnected it should release all pending flush calls with a disconnected error etc. Check Go client.